braintree/webhook_testing_gateway.py

```diff
@@ -452,7 +461,24 @@ def __subscription_charged_successfully_sample_xml(self, id):
                 <add_ons type="array"></add_ons>
                 <discounts type="array"></discounts>
             </subscription>
-        """ % id
+        """ % (id, self.fake_id())
+
+    def __subscription_charged_unsuccessfully_sample_xml(self, id):
+        return """
+            <subscription>
+                <id>%s</id>
+                <transactions type="array">
+                    <transaction>
+                        <id>%s</id>
```

Comment:
In few other cases, we have set the same `id` used for subscription inside transaction as shown here:

https://github.com/rinse-inc/braintree_python/blob/92ac7183870b9a8860cfe948508fcbca0a7dc33e/braintree/webhook_testing_gateway.py#L229-L236

This should remove the need for `random` and will allow you to check its `id`'s concrete value.

Reply:
@bjackson Please make this change.